libgadget/run.c

```diff
@@ -167,8 +167,36 @@ void run(DomainDecomp * ddecomp)
                 endrun(0, "Human triggered termination.\n");
             }
         }
+
+        double new_random_shift[3] = {0};
+        if(NumCurrentTiStep > 0 && is_PM  && All.RandomParticleOffset > 0) {
+            int i;
```

Comment:
Perhaps move this to a function (173-196).
UpdateRandomOffset(All.RandomParticleOffset, All.CurrentParticleOffset, rel_random_shift);